### ISSUE TITLE
Defining a standard way of accessing end-user's storage

### ISSUE THREAD

**Issue by BarracudaX** [solid/solid-oidc#68]
Is there a standard way for acquiring the storage URL of the pod owned by the end-user? Correctly if I am wrong. The primary use of DPoP Access Token in solid is to get resources saved in end-user's pod(s). If we consider a scenario where end-user provide their pod provider(that supports Solid OIDC), after getting the access token, clients have no clue about the storage(s) with which this access token can be used. The answer I got from others was to get the Web ID document and seek for pim:storage predicate(if any). What happens if there is no pim:storage or pim:storage refers to a completely different storage than the one with which the access token can be used? Maybe a claim in the access token that contains the url of storage with which the access token can be used could be helpful in this situation or an endpoint that returns the storge url.

**Comment by acoburn** [comment]
tl;dr -- Storage discovery is beyond the scope of Solid-OIDC.

> Is there a standard way for acquiring the storage URL of the pod owned by the end-user?

Per the [Solid Protocol 4.1 Storage](https://solidproject.org/TR/protocol#storage), a user can make use of the `pim:storage` predicate to discovery a Storage location.

> The answer I got from others was to get the Web ID document and seek for pim:storage predicate(if any). 

Correct, in practice, this typically can be found by dereferencing a user's WebID profile, but there is no requirement that this predicate be present in a user's WebID profile.

> What happens if there is no pim:storage or pim:storage refers to a completely different storage than the one with which the access token can be used?

A user may not wish to publicly disclose the location of a private Storage and there is certainly no reason why a user would be required to disclose this. In this case, surely an app would just ask the user.

All of this, however, is entirely orthogonal to Solid-OIDC. Solid-OIDC is strictly about user and application authentication in Solid. A claim about a storage URL is entirely separate from identity, and the two should not be conflated. In fact, it is entirely likely that an identity provider will not know the location of a Storage URL.

**Comment by elf-pavlik** [comment]
[Solid Application Interoperability](https://solid.github.io/data-interoperability-panel/specification/#datamodel-registry-set) provides a pattern for discovery which also takes care of prior authorization requirements.

You can find introduction video linked from https://github.com/solid/data-interoperability-panel#solid-application-interoperability

**Comment by BarracudaX** [comment]
Okay, I am closing the issue. Thank you for your answers.
